tests/PHPStan/Analyser/data/impure-error-log.php

```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace ImpureErrorLog;
+
+use function PHPStan\Testing\assertType;
+
+$message = 'foo';
+$logfile = 'bar/baz.txt';
+if (!error_log($message, 3, $logfile)) {
+	assertType('bool', error_log($message, 3, $logfile));
+}
```

Comment:
Please always put the code you're testing into an anonymous function. The root scope behaves a little bit differently (see `Scope::canAnyVariableExist()`) and when we are not testing something specific to the root scope I like to avoid issues with that.

Reply:
right, forgot about that. thanks for the headsup.